Support the construction of Taylor form hamiltonian (#6523)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

    The functionality `qml.poly_to_angles` has been also extended to support GQSP.
    [(#6565)](https://github.com/PennyLaneAI/pennylane/pull/6565)

> * Added support to build the vibrational Hamiltonian in Taylor form.

Author: ```suggestion
* Added support to build a vibrational Hamiltonian in Taylor form.
```

> * Added support to build Taylor form Hamiltonians from `VibrationalPES` objects.

Reviewer: ```suggestion
* Added support to build the Taylor vibrational Hamiltonian.
```
